File: arrays-hashing/valid-anagram/validAnagram.js
function validAnagram(s, t) {
  if (s.lenght != t.length) {
    return False;
  }

  charCount = {};

  for (const char of s) {
    charCount[char] = (charCount[char] || 0) + 1;
  }

  for (const char of t) {
    if (!(char in charCount)) {
      return false;
    }
    charCount[char]--;
    if (charCount[char] < 0) {
      return false;
    }
  }

  return true;
}

File: arrays-hashing/valid-anagram/valid_anagram.py
def valid_anagram(s: str, t: str):
    if len(s) != len(t):
        return False

    # Count frequencies in s
    char_count = {}
    for char in s:
        char_count[char] = char_count.get(char, 0) + 1

    # Subtract frequencies based on t
    for char in t:
        if char not in char_count:
            return False
        char_count[char] -= 1
        if char_count[char] < 0:
            return False

    return True
